hydra/core/override_parser/types.py

```diff
@@ -259,8 +259,14 @@ def _convert_value(value: ParsedElementType) -> Optional[ElementType]:
         if isinstance(value, list):
             return [Override._convert_value(x) for x in value]
         elif isinstance(value, dict):
+
+            # Currently only strings are allowed as dictionary keys.
+            def check_str(k: Any) -> str:
+                assert isinstance(k, str)
+                return k
+
```

Comment:
Oh, that's a good point! I know your comment was related to the wording, but it made more sense to me to revert the dict key type to `str` since that's the only type that is actually supported right now (once https://github.com/omry/omegaconf/pull/454 is merged it should be changed to `DictKeyType`, but we don't have that yet). And once I reverted it to `str` I had to fix `mypy` complaints and that was the purpose of `check_str`.

I ended up just removing the type check in 0d745041a, this way we let OmegaConf accept / refuse keys instead of checking it in Hydra.